android/src/main/java/com/google/samples/apps/ibeaconapp/lightbluebean/IBeaconManager.java

```diff
@@ -0,0 +1,73 @@
+package com.google.samples.apps.ibeaconapp.lightbluebean;
+
+import com.google.samples.apps.ibeaconapp.beaconinterface.IBeaconInterface;
+import nl.littlerobots.bean.Bean;
+import nl.littlerobots.bean.BeanDiscoveryListener;
+import nl.littlerobots.bean.BeanManager;
+
+import java.util.*;
+
+public class IBeaconManager implements IBeaconInterface {
+
+    private static volatile IBeaconManager instance;
+    private BeanDiscoveryListener beanDiscoveryListener;
+    private volatile HashMap<Bean, Integer> tempBeansAndRssi = new HashMap<Bean, Integer>();
```

Comment:
This class fields ( beanDiscoveryListener , tempBeansAndRssi , etc) could be considered as class global "state" . It is a good practice to have as less as possible of this "global" state. It is better from memory management perspective(GC couldn't collect this instances until hole IBeaconManager instance could be collected), work in multithreaded environments and general from "error prone" perspective.
I mean - if you could do local variables instead of class fields - it is better to do it.
And this guys - beanDiscoveryListener and tempBeansAndRssi - are local in their nature. You need them and use them in one method only. Lets keep them here.